docs/data/routes/docs/nextjs/data-fetching/component-level-data-fetching/en.md

```diff
@@ -3,4 +3,98 @@ name: component-level-data-fetching
 routeTemplate: ./data/component-templates/article.yml
 title: Component-level data fetching
 ---
-# Component-level data fetching
\ No newline at end of file
+# Component-Level Data Fetching
+
+Next.js provides the ability to pre-render pages at build time using [`getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) or on each request using [`getServerSideProps`](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering). It expects you to fetch all required data in `getStaticProps`/`getServerSideProps`. We provide the ability to fetch page-level data using the [`SitecorePagePropsFactory`](./data-services). 
+
+We can therefore pre-render pages with data fetched on the page level. But what if we want to fetch component-specific data at the component level? 
+
+In the JSS Next.js sample application, we provide a simple way to fetch component-specific data at the component level. 
+
+For this purpose, the `SitecorePagePropsFactory` uses the `ComponentPropsService`. 
+
+## The ComponentPropsService
+
+The `ComponentPropsService ` accepts the following parameters:
+
+- `componentModule` - a function returning a Next.js component using `componentName`. You can find it in `temp/componentFactory`. If not present, generate it by running  `scripts/bootstrap.ts`.
+- `layoutDat`  - Layout Service data for your page.
+- `context` - SSG or SSR Next.js context.
+
+The `ComponentPropsService` goes through the layout service data and looks at all the renderings. To find the components that require data fetching,  the service "spies" on the component using the `rendering.componentName` against the list of component registrations in  `componentFactory`. 
+
+If the component defines and exports the functions `getStaticProps` and `getServerSideProps`, the `ComponentPropsService` executes one of the functions as follows: 
+
+- In SSR mode, it will call `getServerSideProps`.
+- In SSG mode, it will call `getStaticProps`. 
+
+After executing all side-effects, it will store all the data in the format `{ [rendering.uid]: data }`.
+
+## An example with a ContenBlock component
+
+Let's look at an example of a component with side-effects. 
+
+In  `src/components/ContentBlock.tsx`, we define the `getStaticProps` and `getServerSideProps` functions  to fetch the post we want to render. 
```

Comment:
Should we be clear that you only need to implement the functions needed for your solution? If you are doing all SSG, you don't need `getServerSideProps`.